js/bootstrap-select.js

```diff
@@ -2250,7 +2250,11 @@
         // This is useful for smaller menus, where there might be plenty of room
         // below the button without setting dropup, but we can't know
         // the exact height of the menu until createView is called later
-        estimate = liHeight * this.selectpicker.current.data.length + menuPadding.vert;
+        if (this.options.size !== 'auto') {
```

Comment:
you could avoid an `else` with a var holding either `this.options.size` or `this.selectpicker.current.data.length` depending on the value of `this.options.size`. And it would reduce code duplication for the `estimate` var.

Reply:
ok